### ISSUE TITLE
Cannot run genSources in subproject if root project does not have the fabric-loom plugin

### ISSUE THREAD

**Issue by solonovamax** [FabricMC/fabric-loom#308]
When running `genSources` in a subproject, you get the following error:
```
Error: Could not find or load main class net.fabricmc.loom.decompilers.fernflower.FabricForkedFFExecutor
Caused by: java.lang.ClassNotFoundException: net.fabricmc.loom.decompilers.fernflower.FabricForkedFFExecutor
```
This seems to be caused by the commit db1fe88, where for *some reason*, it was decided that it would be a good idea to rely on the root project having the `fabric-loom` gradle plugin.
If you'd like to reproduce this error, you can clone this repo [here](https://github.com/solonovamax/Terra/tree/agnostic) (make sure to check out the right branch), or try and recreate it in a clean environment ig.

If you want me to provide any other stack traces or whatever, feel free to ask.

Tagging #181, as that was the issue that prompted this commit.
(There were no other open issues for this, so this is not a duplicate.)

**Comment by i509VCB** [comment]
~~Would running `:fabric:genSources` work? I do have some experience with this from the luckperms PR I have made~~

Edit: I misread the message lol, you said you ran it in subproject

Ideally this could be fixed by checking if loom is applied to root project and or any other projects. If loom is only applied to one project, then that should run fine with a small modification, but if it's two sub projects and not root then modmuss would need to figure out whats appropriate there.

**Comment by Tigermouthbear** [comment]
@solonovamax can you make a pr with your fix? I'm getting the same error

**Comment by modmuss50** [comment]
The fix was merged in the (currenly unstable) 0.6 branch: https://github.com/FabricMC/fabric-loom/commit/c96396fb2732e76ee460be622b8ba928ee95f24e
